RipCheck/Tracks/DrumsTrack/DrumsTrack.cs

```diff
@@ -53,6 +56,34 @@ public DrumsTrack(TrackChunk track, TempoMap _tempoMap, string instrument, Check
             }
         }
 
+        public Warnings CheckExpertPlus()
+        {
+            var warnings = new Warnings();
+
+            var diff = notes[Difficulty.Expert];
+            IEnumerable<INote> kicks = diff.Where((note) => note.Note == (byte)DrumsFretColour.Kick);
+            IEnumerable<INote> doubleKicks = diff.Where((note) => note.Note == (byte)DrumsFretColour.DoubleKick);
+
+            foreach (var kick in kicks)
+            {
+                bool found = false;
+                foreach (var doubleKick in doubleKicks)
+                {
+                    if (doubleKick.Position == kick.Position)
+                    {
+                        found = true;
+                    }
+                }
+
```

Comment:
I'm not keen on this double loop, the code would be clearer andbetter performing if you did something like form a set of all the double kick positions before the loop and then check membership of that.

Reply:
Something like this?

```cs
var kicks = diff.Where((note) => note.Note == (byte)DrumsFretColour.Kick);
var doubleKickPositions = diff.Where((note) => note.Note == (byte)DrumsFretColour.DoubleKick).Select((note) => note.Position);

foreach (var kick in kicks)
{
    if (!doubleKickPositions.Contains(kick.Position))
    {
        warnings.AddTimed($"Found Expert kick with no corresponding Expert+ kick on {name}", kick.Position, tempoMap);
    }
}
```

I'm not sure this is really more performant, it's just replacing an explicit loop with an enumerable which mostly does the same thing. It's definitely cleaner though.

Reply:
Yes, except you throw a `ToHashSet` at the end of your expression for `doubleKickPositions`, which helps the performance since `Contains` is O(1) on `HashSet`.

Reply:
Alrighty, will do.